proseco/guide.py

```diff
@@ -109,7 +134,154 @@ def reject(self, reject):
+            dist = np.linalg.norm([self.stars['yang'] - monitor['yang'],
+                                   self.stars['zang'] - monitor['zang']], axis=0)
+            idxs = np.flatnonzero(dist < 2.0)
+            if len(idxs) == 0:
+                raise BadMonitorError('no acceptable AGASC star within '
+                                      '2 arcsec of monitor position')
+            elif len(idxs) == 1:
+                self.include_ids.append(self.stars['id'][idxs[0]])
+            else:
+                # Do something better like choose the brightest?  Probably in all realistic
+                # cases this just doesn't happen.
+                raise BadMonitorError('multiple guide candidates within '
+                                      '2 arcsec of monitor position')
+
+    def process_monitors_post(self):
+        """Post-processing of monitor windows.
+
+        - Clean up fake stars from stars and cand_guides
+        - Restore the dark current map
+        - Set type, sz, res, dim columns for relevant entries to reflect
+          status as monitor or guide-from-monitor in the guides catalog.
+        """
+        if self.monitors is None:
+            return
+
+        # Mon window processing munges the dark cal to impose a keep-out zone.
+        # Put the dark current back to the standard one if possible
+        if self.acqs is not None:
+            self.dark = self.acqs.dark
+
+        # Process monitor windows according to function
+        mon_id = 1000
+        for monitor in self.monitors:
+            if monitor['function'] == MonFunc.GUIDE:
+                dist = np.linalg.norm([self['yang'] - monitor['yang'],
+                                       self['zang'] - monitor['zang']], axis=0)
+                ok = dist < 2.0
+                if np.count_nonzero(ok) != 1:
+                    # Should never happen.
+                    raise RuntimeError('code logic error: expected exactly one '
+                                       'match to a Guide from MON request')
+                self['sz'][ok] = '8x8'
+                self['type'][ok] = 'GFM'  # Guide From Monitor, changed in merge_catalog
+
+            elif monitor['function'] in (MonFunc.MON_FIXED, MonFunc.MON_TRACK):
+                # Undo the temporary reduction in n_guide for MON windows.
+                # n_guide refers to the number of GUI + MON in the guide catalog.
+                self.n_guide += 1
+
+                # Make a stub row for a MON entry using zero everywhere. This
+                # also works for str (giving '0').
+                mon = {col.name: col.dtype.type(0) for col in self.itercols()}
+                # These type codes get fixed later in merge_catalog
+                mon['type'] = 'MFX' if monitor['function'] == MonFunc.MON_FIXED else 'MTR'
+                mon['sz'] = '8x8'
+                mon['dim'] = -999  # Obviously bad value for DTS, gets fixed later.
+                mon['res'] = 0
+                mon['halfw'] = 20
```

Comment:
I haven't dug up the manual but note that starcheck looks to have options to support 10 15 and 20 based on IMGSZ

https://github.com/sot/starcheck/blob/f98cd278591714b84d4c69542c235f61b633eb01/starcheck/src/lib/Ska/Starcheck/Obsid.pm#L763